src/ipc-handlers.ts

```diff
@@ -389,6 +394,19 @@ export async function startServer(
 
 	await keepSqliteIntegrationUpdated( server.details.path );
 
+	// Handle custom domain if necessary
+	if ( server.details.useCustomDomain && server.details.customDomain ) {
+		try {
+			await addDomainToHosts( server.details.customDomain, server.details.port );
+			console.log(
+				`Domain ${ server.details.customDomain } added to hosts file for port ${ server.details.port }`
+			);
+		} catch ( error ) {
+			console.error( 'Failed to setup custom domain:', error );
+			// Continue even if custom domain setup fails - the site will still work with localhost:PORT
```

Comment:
We no longer show the site port in the UI, though. And wouldn't WordPress redirect to the custom domain..?